scripts/abci.js

```diff
@@ -31,24 +35,56 @@ const errorHandlerFactory = require('../lib/errorHandlerFactory');
   });
 
   logger.info('Connecting to Core');
-  const detectStandaloneRegtestMode = container.resolve('detectStandaloneRegtestMode');
-
-  const waitForDmlActivated = container.resolve('waitForDmlActivated');
-
-  logger.info('Waiting for DML being activated...');
-
-  await waitForDmlActivated();
 
+  const detectStandaloneRegtestMode = container.resolve('detectStandaloneRegtestMode');
   const isStandaloneRegtestMode = await detectStandaloneRegtestMode();
 
+  /**
+   * Make sure Core is synced
+   */
+
   if (!isStandaloneRegtestMode) {
     const waitForCoreSync = container.resolve('waitForCoreSync');
     await waitForCoreSync((currentBlockHeight, currentHeaderNumber) => {
       logger.info(
         `waiting for Core to finish sync ${currentBlockHeight}/${currentHeaderNumber}...`,
       );
     });
+  }
+
+  /**
+   * Connect to Core ZMQ socket
+   */
+
+  const coreZMQClient = container.resolve('coreZMQClient');
+  const coreZMQConnectionRetries = container.resolve('coreZMQConnectionRetries');
+
+  try {
+    await coreZMQClient.connect({
+      maxRetries: coreZMQConnectionRetries,
+    });
+  } catch (e) {
+    const error = new Error(`Can't connect to Core ZMQ socket: ${e.message}`);
+
+    await errorHandler(error);
+  }
+
+  // By default will try to reconnect so we just log when this happen
+  coreZMQClient.on('connect', () => {
```

Comment:
I would suggest setting the listener before the connect method call as we call it with await

Reply:
Make sense!